nautobot/extras/filters/__init__.py

```diff
@@ -599,8 +596,9 @@ class JobHookFilterSet(BaseFilterSet):
         choices=ChangeLoggedModelsQuery().get_choices,
     )
     job = NaturalKeyOrPKMultipleChoiceFilter(
+        to_field_name="name",
         queryset=Job.objects.all(),
-        label="Job (slug or ID)",
+        label="Job (name or ID)",
     )
```

Comment:
Since `name` is not unique, can this be anything but `id`?

This might be a case for making `class_path` cached on the JobModel.

Reply:
See this will have to change in: https://github.com/nautobot/nautobot/pull/3840/files#r1214707739